Classes/Emogrifier.php

```diff
@@ -294,16 +300,18 @@ public function emogrify()
         // supports XPath 1.0, lower-case() isn't available to us. We've thus far only set attributes to lowercase,
         // not attribute values. Consequently, we need to translate() the letters that would be in 'NONE' ("NOE")
         // to lowercase.
-        $nodesWithStyleDisplayNone = $xpath->query(
-            '//*[contains(translate(translate(@style," ",""),"NOE","noe"),"display:none")]'
-        );
-        // The checks on parentNode and is_callable below ensure that if we've deleted the parent node,
-        // we don't try to call removeChild on a nonexistent child node
-        if ($nodesWithStyleDisplayNone->length > 0) {
-            /** @var \DOMNode $node */
-            foreach ($nodesWithStyleDisplayNone as $node) {
-                if ($node->parentNode && is_callable(array($node->parentNode,'removeChild'))) {
-                    $node->parentNode->removeChild($node);
+        if ($this->isRemoveDisplayNoneEnabled) {
+            $nodesWithStyleDisplayNone = $xpath->query(
```

Comment:
Please let's move the code within the IF to a separate, well-named private method.